File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> missing;
        vector<int> temp(nums.size(),0);
        for(int i = 0 ; i < nums.size() ; i++){
            temp[nums[i]-1]++;
        }  
        for(int i = 0 ; i < nums.size() ; i++){
            if(temp[i] == 0)
                missing.push_back(i+1);
        }
        return missing;  
    }
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int[] freq = new int[nums.length] ; // initalizes to 0
        List<Integer> ans = new ArrayList<>() ;
        for(int i = 0 ; i < nums.length ; i++){
            freq[nums[i] - 1 ]++ ;
        }
        for(int i = 0 ; i < freq.length ; i++){
            if(freq[i] == 0)
                ans.add(i+1);
        }
        return ans ;
    }
}